aten/src/ATen/native/cuda/ReduceSumProdKernel.cu

```diff
@@ -162,8 +186,9 @@ static void sum_kernel_cuda(TensorIterator& iter){
 
 static void nansum_kernel_cuda(TensorIterator& iter) {
   auto general_dispatcher = [](TensorIterator& iter) {
-    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "nansum_cuda", [&]() {
-      nansum_functor<scalar_t>{}(iter);
+    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(kComplexHalf,
+      iter.dtype(), "nansum_cuda", [&]() {
+        nansum_functor<scalar_t>{}(iter);
```

Comment:
Any estimation how much this affects compile time and binary size?

Reply:
Hi, @malfet!

> Any estimation how much this affects compile time and binary size?

I usually take 45-60 mins to compile in my system. The last time I compiled, I got the following:
```
real	47m6.068s
user	537m54.813s
sys	17m20.032s
```
<details>
<summary>My System Info</summary>

<p>

```python
Collecting environment information...
PyTorch version: 2.0.0a0+git295a5e2
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: 10.0.0-4ubuntu1 
CMake version: version 3.22.1
Libc version: glibc-2.31

Python version: 3.10.8 (main, Nov 24 2022, 14:13:03) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-58-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 11.6.124
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1650 Ti
Nvidia driver version: 470.161.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False

Versions of relevant libraries:
[pip3] mypy==0.960
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.23.1
[pip3] torch==2.0.0a0+gitf6a3034
[conda] mkl                       2022.1.0           hc2b9512_224  
[conda] mkl-include               2022.1.0           h06a4308_224  
[conda] numpy                     1.23.1                   pypi_0    pypi
[conda] torch                     2.0.0a0+gitc805b2b          pypi_0    pypi
```

</p>
</details>

Reply:
Build time of whole build seems to have a lot of variance.

One thing you can do is build PyTorch first and then make a small change (maybe add a space) in `ReduceSumProdKernel.cu` to trigger incremental build so that only `ReduceSumProdKernel.cu` and linking will be triggered. This will provide more granular time required to compile the file.

Other option (if you are using `ninja` for building) is to use https://github.com/nico/ninjatracing and check how long this particular file takes.

Reply:
I added a line to check the build time, as you suggested. I got:
```python
real	0m57.989s
user	0m55.940s
sys	0m4.402s
```